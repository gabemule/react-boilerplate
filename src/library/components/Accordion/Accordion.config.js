module.exports = {
  name: 'Accordion',
  class: 'ds-accordion',
  props: {
    items: {
      type: Array,
      default: []
    }
  }
}
