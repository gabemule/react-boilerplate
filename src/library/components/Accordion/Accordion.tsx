import React, { useState } from 'react'
// import withCustomProps from '@/library/decorators/withCustomProps'
import PropTypes from "prop-types";

import DsAccordionItem from './AccordionItem'
import { DsAccordionItemProps } from '@/library/types/types'

// config
import AccordionConfig from "./Accordion.config";

const DsAccordion = ({ items }: { items: DsAccordionItemProps[] }) => {
  return (
    <ul className={AccordionConfig.class}>
      {items.map((item, index) => {
        return <DsAccordionItem key={index} index={index} title={item.title} disabled={item.disabled} content={item.content} />
      }
      )}
    </ul>
  )
}

DsAccordion.propTypes = {
  items: PropTypes.array
};

DsAccordion.defaultProps = {
  items: []
}

export default DsAccordion;
