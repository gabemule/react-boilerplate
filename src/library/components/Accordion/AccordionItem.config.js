module.exports = {
  name: 'AccordionItem',
  class: 'ds-accordion-item',
  props: {
    title: {
      type: String,
      default: ''
    },
    content: {
      type: String,
      default: ''
    },
    disabled: {
      type: Boolean,
      default: false
    }
  }
}
