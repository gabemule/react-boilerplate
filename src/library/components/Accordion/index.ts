import DsAccordion from "./Accordion"
import { DsAccordionItemProps }from "@/library/types/types";

export { DsAccordion }

export type { DsAccordionItemProps }
