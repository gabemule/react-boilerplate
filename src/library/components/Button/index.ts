import DsButton from "./Button";
import { DsButtonProps }from "./Button";

export { DsButton };
export type { DsButtonProps }
