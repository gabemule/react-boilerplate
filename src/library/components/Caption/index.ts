import DsCaption from "./Caption";
import { DsCaptionProps } from "./Caption"

export { DsCaption };
export type { DsCaptionProps }
