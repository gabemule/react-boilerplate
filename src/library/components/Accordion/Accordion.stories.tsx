import React from "react";
import { Meta, Story } from "@storybook/react";

import { ThemeProvider, ThemeProviderProps } from "@/library/components/ThemeProvider";
import { DsAccordion } from "@/library/components/Accordion";
import { DsAccordionItemProps } from '@/library/types/types'

const componentProps: Meta = {
  title: "Interação Básica/Accordion/Accordion",
  component: DsAccordion,
  argTypes: {
  }
};

export default componentProps;

const defaultItems = [
  {
    title: 'Accordion 1',
    content: 'lorem ipsum dolor sit amet',
    disabled: false
  },
  {
    title: 'Accordion 2',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Consequuntur, saepe tempore, dolorum nostrum earum ex porro sapiente illum asperiores aliquid repellendus exercitationem, a voluptates. Eveniet aliquam, nisi magni officiis delectus nam voluptates. Ratione quos vel, est culpa odio, officiis eius tempora itaque quasi inventore obcaecati quidem et accusamus ea cupiditate iusto sit. Impedit voluptates error aliquam perspiciatis ipsa totam corporis consectetur nihil odio assumenda voluptatem quidem aspernatur itaque eaque repudiandae consequuntur nisi inventore vero ab quasi, hic repellendus est reprehenderit? Autem nesciunt veniam nihil tempora magni error facere, aut repellat! Excepturi, non. Magnam voluptatem harum eum sit sint. Perferendis, vel!',
    disabled: false
  },
  {
    title: 'Accordion 3',
    content: 'lorem ipsum dolor sit amet',
    disabled: false
  },
  {
    title: 'Accordion 4',
    content: 'lorem ipsum dolor sit amet',
    disabled: true
  },
  {
    title: 'Accordion 5',
    content: 'lorem ipsum dolor sit amet',
    disabled: false
  }
]


const Template: Story<ThemeProviderProps & DsAccordionItemProps> = (args) => (
  <ThemeProvider library={"core"} theme={"light"}>
    <DsAccordion {...args} items={defaultItems} />
  </ThemeProvider>
);

export const Default = Template.bind({});
Default.args = {};
