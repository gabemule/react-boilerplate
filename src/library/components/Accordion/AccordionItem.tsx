import React, { useState } from 'react'
// import withCustomProps from '@/library/decorators/withCustomProps'
import PropTypes from "prop-types";

import { DsAccordionItemProps } from '@/library/types/types'
import { DsIcon } from '@/library/components/Icon';
import { DsParagraph } from '@/library/components/Paragraph';

// config
import AccordionItemConfig from "./AccordionItem.config";

const DsAccordionItem = ({ index, title, disabled, content, className }: DsAccordionItemProps) => {
  const [clicked, setClicked] = useState(index === 0);

  const accordionItemDisabled = disabled && `${AccordionItemConfig.class}--disabled`;
  const accordionItemIsOpen = clicked && `${AccordionItemConfig.class}--active`;

  const handleToggle = () => {
    setClicked((prevState) => !prevState);
  };

  return (
    <li
      className={[AccordionItemConfig.class, accordionItemDisabled, accordionItemIsOpen, className && className].filter(classes => classes).join(" ")}
      data-testid="accordionItem"
    >
      <button className={`${AccordionItemConfig.class}__button`} disabled={disabled} onClick={handleToggle}>
        <DsParagraph type="small">
          {title}
        </DsParagraph>
        <DsIcon image={disabled ? 'block' : 'keyboard-arrow-down'} color={disabled ? 'neutral-40' : 'neutral-90'} />
      </button>
      {clicked &&
        <div className="ds-accordion-item__content">
          {content}
        </div>}
    </li>
  )
}

DsAccordionItem.propTypes = {
  index: PropTypes.number,
  title: PropTypes.string,
  content: PropTypes.string,
  disabled: PropTypes.bool,
  className: PropTypes.string,
}

DsAccordionItem.defaultProps = {
  title: "",
  content: "",
  disabled: false
}

export default DsAccordionItem;
