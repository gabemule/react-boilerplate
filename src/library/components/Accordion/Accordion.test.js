import React from 'react'
import { shallow, render } from 'enzyme'

// components
import { DsAccordion } from '@/library/components/Accordion'

// config
import AccordionConfig from './Accordion.config'

const AccordionClass = AccordionConfig.class
const mockedItems = [
  {
    title: 'Accordion 1',
    content: 'lorem ipsum dolor sit amet'
  },
  {
    title: 'Accordion 2',
    content: 'lorem ipsum dolor sit amet'
  }
]

describe('Accordion Component', () => {
  it('Matches Snapshot', () => {
    const wrapper = render(<DsAccordion items={mockedItems} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component HTML', () => {
    const wrapper = shallow(<DsAccordion items={mockedItems} />)
    expect(wrapper).toBeTruthy()
    expect(wrapper.find(AccordionClass)).toBeTruthy()
  })
})
