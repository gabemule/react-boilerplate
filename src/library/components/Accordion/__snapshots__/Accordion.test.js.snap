// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component Matches Snapshot 1`] = `
LoadedCheerio {
  "0": <ul
    class="ds-accordion"
  >
    <li
      class="ds-accordion-item ds-accordion-item--active"
      data-testid="accordionItem"
    >
      <button
        class="ds-accordion-item__button"
      >
        <p
          class="ds-paragraph ds-paragraph--small  with-custom-props"
          data-testid="text-text"
        >
          Accordion 1
        </p>
        <div
          class="ds-icon ds-icon--keyboard-arrow-down ds-icon__color--neutral-90"
          data-testid="text-icon"
        />
      </button>
      <div
        class="ds-accordion-item__content"
      >
        lorem ipsum dolor sit amet
      </div>
    </li>
    <li
      class="ds-accordion-item"
      data-testid="accordionItem"
    >
      <button
        class="ds-accordion-item__button"
      >
        <p
          class="ds-paragraph ds-paragraph--small  with-custom-props"
          data-testid="text-text"
        >
          Accordion 2
        </p>
        <div
          class="ds-icon ds-icon--keyboard-arrow-down ds-icon__color--neutral-90"
          data-testid="text-icon"
        />
      </button>
    </li>
  </ul>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
